### ISSUE TITLE
Delete message action causes crash 

### ISSUE THREAD

**Issue by srgray** [GetStream/stream-chat-swift#1295]
## What did you do?
Set the new `ChatClientConfig.deletedMessagesVisibility` = `.alwaysHidden`
Delete a message in a chat channel

## What did you expect to happen?
Message to be deleted and not shown in UI

## What happened instead?
App crash:
Swift/ContiguousArrayBuffer.swift:580: Fatal error: Index out of range
`LazyCachedMapCollection.swift` line 39

## GetStream Environment
**GetStream Chat version:** 4.0.0-beta.7
**GetStream Chat frameworks:** StreamChat, StreamChatUI

**Comment by b-onc** [comment]
Hello @srgray ,

I can't reproduce the issue. Is the issue consistently reproducible for you? Can you share the app id (not API key, App id) and channel id this happens?

**Comment by tbarbugli** [comment]
@srgray if possible a stacktrace would be super useful as well to quickly resolve this

**Comment by srgray** [comment]
Yes, I am able to reproduce this very consistently.
AppID: `82241`
ChannelID: `livestream:71c386ae-2788-46d4-8545-172de370dfcd` (but I have reproduced this on several channels)
Stack Trace:
```
(lldb) bt
* thread #1, queue = 'com.apple.main-thread', stop reason = Fatal error: Index out of range
    frame #0: 0x00007fff2f36ab50 libswiftCore.dylib`_swift_runtime_on_report
    frame #1: 0x00007fff2f3e4e8c libswiftCore.dylib`_swift_stdlib_reportFatalErrorInFile + 204
    frame #2: 0x00007fff2f07d8ee libswiftCore.dylib`closure #1 (Swift.UnsafeBufferPointer<Swift.UInt8>) -> () in closure #1 (Swift.UnsafeBufferPointer<Swift.UInt8>) -> () in closure #1 (Swift.UnsafeBufferPointer<Swift.UInt8>) -> () in Swift._assertionFailure(_: Swift.StaticString, _: Swift.StaticString, file: Swift.StaticString, line: Swift.UInt, flags: Swift.UInt32) -> Swift.Never + 478
    frame #3: 0x00007fff2f07d509 libswiftCore.dylib`closure #1 (Swift.UnsafeBufferPointer<Swift.UInt8>) -> () in closure #1 (Swift.UnsafeBufferPointer<Swift.UInt8>) -> () in Swift._assertionFailure(_: Swift.StaticString, _: Swift.StaticString, file: Swift.StaticString, line: Swift.UInt, flags: Swift.UInt32) -> Swift.Never + 105
    frame #4: 0x00007fff2f07d2a9 libswiftCore.dylib`closure #1 (Swift.UnsafeBufferPointer<Swift.UInt8>) -> () in Swift._assertionFailure(_: Swift.StaticString, _: Swift.StaticString, file: Swift.StaticString, line: Swift.UInt, flags: Swift.UInt32) -> Swift.Never + 121
    frame #5: 0x00007fff2f07cf6c libswiftCore.dylib`Swift._assertionFailure(_: Swift.StaticString, _: Swift.StaticString, file: Swift.StaticString, line: Swift.UInt, flags: Swift.UInt32) -> Swift.Never + 428
    frame #6: 0x00007fff2f0c18bb libswiftCore.dylib`Swift.ContiguousArray.subscript.getter : (Swift.Int) -> τ_0_0 + 123
    frame #7: 0x000000010a213997 StreamChat`LazyCachedMapCollection.subscript.getter(position=0, self=StreamChat.LazyCachedMapCollection<StreamChat._ChatMessage<CrowdNoise.CNExtraData>> @ 0x00007ffee9ac76c0) at LazyCachedMapCollection.swift:39:37
    frame #8: 0x000000010ab975ed StreamChatUI`_ChatMessageListVC.scrollOverlay(overlay=0x00007fccf7ce2b90, indexPath=Foundation.IndexPath @ 0x00007ffee9ac7a40, self=0x00007fccf921b800) at ChatMessageListVC.swift:255:53
    frame #9: 0x000000010aba4b7e StreamChatUI`protocol witness for ChatMessageListScrollOverlayDataSource.scrollOverlay(_:textForItemAt:) in conformance _ChatMessageListVC<A> at <compiler-generated>:0
    frame #10: 0x000000010ab8b27a StreamChatUI`closure #1 in ChatMessageListScrollOverlayView.listView.didset(tb=0x00007fccf896e200, self=0x00007fccf7ce2b90) at ChatMessageListScrollOverlayView.swift:50:52
    frame #11: 0x000000010ab8b69f StreamChatUI`thunk for @escaping @callee_guaranteed (@guaranteed UITableView, @in_guaranteed NSKeyValueObservedChange<CGPoint>) -> () at <compiler-generated>:0
    frame #12: 0x00007fff53de5dae libswiftFoundation.dylib`closure #1 (__C.NSObject, Foundation.NSKeyValueObservedChange<Any>) -> () in Foundation._KeyValueCodingAndObserving.observe<τ_0_0>(_: Swift.KeyPath<τ_0_0, τ_1_0>, options: __C.NSKeyValueObservingOptions, changeHandler: (τ_0_0, Foundation.NSKeyValueObservedChange<τ_1_0>) -> ()) -> Foundation.NSKeyValueObservation + 318
    frame #13: 0x00007fff53de5e25 libswiftFoundation.dylib`partial apply forwarder for closure #1 (__C.NSObject, Foundation.NSKeyValueObservedChange<Any>) -> () in Foundation._KeyValueCodingAndObserving.observe<τ_0_0>(_: Swift.KeyPath<τ_0_0, τ_1_0>, options: __C.NSKeyValueObservingOptions, changeHandler: (τ_0_0, Foundation.NSKeyValueObservedChange<τ_1_0>) -> ()) -> Foundation.NSKeyValueObservation + 37
    frame #14: 0x00007fff53de93d6 libswiftFoundation.dylib`function signature specialization <Arg[0] = Dead, Arg[3] = Dead> of Foundation.NSKeyValueObservation.Helper._swizzle_me_observeValue(forKeyPath: Swift.Optional<Swift.String>, of: Swift.Optional<Any>, change: Swift.Optional<Swift.Dictionary<__C.NSString, Any>>, context: Swift.Optional<Swift.UnsafeMutableRawPointer>) -> () + 1510
    frame #15: 0x00007fff53de57ec libswiftFoundation.dylib`@objc Foundation.NSKeyValueObservation.Helper._swizzle_me_observeValue(forKeyPath: Swift.Optional<Swift.String>, of: Swift.Optional<Any>, change: Swift.Optional<Swift.Dictionary<__C.NSString, Any>>, context: Swift.Optional<Swift.UnsafeMutableRawPointer>) -> () + 204
    frame #16: 0x00007fff207c2be4 Foundation`NSKeyValueNotifyObserver + 329
    frame #17: 0x00007fff207c6337 Foundation`NSKeyValueDidChange + 439
    frame #18: 0x00007fff207c5cb7 Foundation`-[NSObject(NSKeyValueObservingPrivate) _changeValueForKeys:count:maybeOldValuesDict:maybeNewValuesDict:usingBlock:] + 741
    frame #19: 0x00007fff207c651a Foundation`-[NSObject(NSKeyValueObservingPrivate) _changeValueForKey:key:key:usingBlock:] + 68
    frame #20: 0x00007fff207c0297 Foundation`_NSSetPointValueAndNotify + 304
    frame #21: 0x00007fff24ba7ee5 UIKitCore`-[UIScrollView _adjustContentOffsetIfNecessary] + 53
    frame #22: 0x00007fff248d6289 UIKitCore`-[UITableView _restoreOrAdjustContentOffsetWithRowCount:initialContentInsetTop:] + 493
    frame #23: 0x00007fff248c3900 UIKitCore`-[UITableView _updateVisibleCellsNow:] + 4213
    frame #24: 0x00007fff248bc419 UIKitCore`-[UITableView _setupCellAnimations] + 46
    frame #25: 0x00007fff248d9ec6 UIKitCore`-[UITableView beginUpdates] + 105
    frame #26: 0x00007fff248da1a6 UIKitCore`-[UITableView _performBatchUpdates:withContext:completion:] + 218
    frame #27: 0x00007fff248da2af UIKitCore`-[UITableView performBatchUpdates:completion:] + 97
    frame #28: 0x000000010aba8a39 StreamChatUI`_ChatMessageListView.updateMessages(changes=1 value, completion=nil, self=0x00007fccf896e200) at ChatMessageListView.swift:188:9
    frame #29: 0x000000010ab9a956 StreamChatUI`_ChatMessageListVC.updateMessages(changes=1 value, completion=nil, self=0x00007fccf921b800) at ChatMessageListVC.swift:350:18
  * frame #30: 0x0000000106405d69 CrowdNoise`CNChatMessageListVC.updateMessages(changes=1 value, completion=nil, self=0x00007fccf921b800) at CNChatMessageListVC.swift:59:15
    frame #31: 0x000000010ab9e107 StreamChatUI`_ChatMessageListVC.channelController(channelController=0x0000600002e89ec0, changes=1 value, self=0x00007fccf921b800) at ChatMessageListVC.swift:479:9
    frame #32: 0x000000010aba498e StreamChatUI`protocol witness for _ChatChannelControllerDelegate.channelController(_:didUpdateMessages:) in conformance _ChatMessageListVC<A> at <compiler-generated>:0
    frame #33: 0x000000010a0970f9 StreamChat`closure #3 in AnyChannelControllerDelegate.init<ExtraData>($0=0x0000600002e89ec0, $1=1 value, delegate=0x00007fccf921b800) at ChannelController.swift:1297:73
    frame #34: 0x000000010a0968d6 StreamChat`AnyChannelControllerDelegate.channelController(controller=0x0000600002e89ec0, changes=1 value, self=0x000060000303ca80) at ChannelController.swift:1273:9
    frame #35: 0x000000010a086eac StreamChat`closure #1 in closure #2 in closure #1 in _ChatChannelController.setMessagesObserver($0=0x000060000303ca80, self=0x0000600002e89ec0, changes=1 value) at ChannelController.swift:326:24
    frame #36: 0x000000010a085fbf StreamChat`thunk for @escaping @callee_guaranteed (@guaranteed AnyChannelControllerDelegate<A>) -> () at <compiler-generated>:0
    frame #37: 0x000000010a2db1c0 StreamChat`MulticastDelegate.invoke(action=0x10a0a4700, self=StreamChat.MulticastDelegate<StreamChat.AnyChannelControllerDelegate<CrowdNoise.CNExtraData>> @ 0x00007ffee9ac9470) at MulticastDelegate.swift:15:13
    frame #38: 0x000000010a1bbcad StreamChat`closure #1 in DelegateCallable.delegateCallback(self=0x0000600002e89ec0, callback=0x10a0a4700) at DataController.swift:81:36
    frame #39: 0x000000010a183b9a StreamChat`Controller.callback(action=0x000000010a1bbd90, self=0x0000600002e89ec0) at Controller.swift:22:17
    frame #40: 0x000000010a08a810 StreamChat`protocol witness for DelegateCallable.callback(_:) in conformance _ChatChannelController<A1> at <compiler-generated>:0
    frame #41: 0x000000010a1bbacb StreamChat`DelegateCallable.delegateCallback(callback=0x10a0a4700, self=0x0000600002e89ec0) at DataController.swift:80:14
    frame #42: 0x000000010a086e1c StreamChat`closure #2 in closure #1 in _ChatChannelController.setMessagesObserver(changes=1 value, self=0x0000600002e89ec0) at ChannelController.swift:325:22
    frame #43: 0x000000010a21b7a4 StreamChat`closure #1 in ListDatabaseObserver.onChange.didset($0=1 value, self=0x0000600001f72d61) at ListDatabaseObserver.swift:103:30
    frame #44: 0x000000010a22213f StreamChat`ListChangeAggregator.controllerDidChangeContent(controller=0x0000600002a9cdc0, self=0x00006000001506c0) at ListDatabaseObserver.swift:345:20
    frame #45: 0x000000010a2227e4 StreamChat`@objc ListChangeAggregator.controllerDidChangeContent(_:) at <compiler-generated>:0
    frame #46: 0x00007fff2526afc9 CoreData`__82-[NSFetchedResultsController(PrivateMethods) _core_managedObjectContextDidChange:]_block_invoke + 7096
    frame #47: 0x00007fff250d323a CoreData`developerSubmittedBlockToNSManagedObjectContextPerform + 154
    frame #48: 0x00007fff250d3121 CoreData`-[NSManagedObjectContext performBlockAndWait:] + 197
    frame #49: 0x00007fff252693ff CoreData`-[NSFetchedResultsController _core_managedObjectContextDidChange:] + 110
    frame #50: 0x00007fff20363882 CoreFoundation`__CFNOTIFICATIONCENTER_IS_CALLING_OUT_TO_AN_OBSERVER__ + 12
    frame #51: 0x00007fff20363851 CoreFoundation`___CFXRegistrationPost_block_invoke + 49
    frame #52: 0x00007fff20362f81 CoreFoundation`_CFXRegistrationPost + 454
    frame #53: 0x00007fff203628ef CoreFoundation`_CFXNotificationPost + 796
    frame #54: 0x00007fff207e9027 Foundation`-[NSNotificationCenter postNotificationName:object:userInfo:] + 59
    frame #55: 0x00007fff25170d9c CoreData`-[NSManagedObjectContext _postContextDidMergeChangesNotificationWithUserInfo:] + 760
    frame #56: 0x00007fff2516f89f CoreData`-[NSManagedObjectContext _createAndPostChangeNotification:deletions:updates:refreshes:deferrals:wasMerge:] + 1994
    frame #57: 0x00007fff251701d5 CoreData`-[NSManagedObjectContext _postRefreshedObjectsNotificationAndClearList] + 206
    frame #58: 0x00007fff25161200 CoreData`-[NSManagedObjectContext _processRecentChanges:] + 112
    frame #59: 0x00007fff250cbb5d CoreData`-[NSManagedObjectContext _mergeChangesFromDidSaveDictionary:usingObjectIDs:] + 5013
    frame #60: 0x00007fff250ca787 CoreData`-[NSManagedObjectContext mergeChangesFromContextDidSaveNotification:] + 410
    frame #61: 0x00007fff250d323a CoreData`developerSubmittedBlockToNSManagedObjectContextPerform + 154
    frame #62: 0x000000010d06474e libdispatch.dylib`_dispatch_client_callout + 8
    frame #63: 0x000000010d072b3f libdispatch.dylib`_dispatch_main_queue_callback_4CF + 1152
    frame #64: 0x00007fff203908f8 CoreFoundation`__CFRUNLOOP_IS_SERVICING_THE_MAIN_DISPATCH_QUEUE__ + 9
    frame #65: 0x00007fff2038b169 CoreFoundation`__CFRunLoopRun + 2781
    frame #66: 0x00007fff2038a1a7 CoreFoundation`CFRunLoopRunSpecific + 567
    frame #67: 0x00007fff2b874d85 GraphicsServices`GSEventRunModal + 139
    frame #68: 0x00007fff246c14df UIKitCore`-[UIApplication _run] + 912
    frame #69: 0x00007fff246c639c UIKitCore`UIApplicationMain + 101
    frame #70: 0x00000001062e205b CrowdNoise`main at AppDelegate.swift:22:7
    frame #71: 0x00007fff2025abbd libdyld.dylib`start + 1
```

**Comment by b-onc** [comment]
Hello @srgray , thanks for all the info!

We've managed to pinpoint the issue and was able to reproduce. A fix PR is opened as #1298 , can you check it out?

**Comment by srgray** [comment]
@b-onc thanks! I think that change looks good. Just a note based on the code comment though: I am able to trigger this crash when the channel contains any number of messages (and not just 1). But I think that code change handles that case, too, if I understand it correctly.

**Comment by srgray** [comment]
This has been fixed on the latest version 4.0.0-beta.8. Thanks!
